code/datums/components/keep_me_secure.dm

```diff
@@ -0,0 +1,66 @@
+/**
+ * ### Keep Me Secure component!
+ *
+ * Component that attaches to items, invoking a function to react when left unmoved and unsecured for too long.
+ * Used for Nuclear Authentication Disks, and whiny plushy as an example (which changes sprites depending on whether it considers itself secure.)
+ */
+/datum/component/keep_me_secure
+	/// callback for the parent being secure
+	var/datum/callback/secured_callback
+	/// callback for the parent being unsecured
+	var/datum/callback/unsecured_callback
+
+	/// The last secure location the parent was at.
+	var/turf/last_secured_location
+	/// The last world time the parent moved.
+	var/last_move
+
+/datum/component/keep_me_secure/Initialize(secured_callback, unsecured_callback)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.secured_callback = secured_callback
+	src.unsecured_callback = unsecured_callback
+
+/datum/component/keep_me_secure/RegisterWithParent()
+	last_move = world.time
+	START_PROCESSING(SSobj, src)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, PROC_REF(on_examine))
+
+/datum/component/keep_me_secure/UnregisterFromParent()
+	STOP_PROCESSING(SSobj, src)
+	UnregisterSignal(parent, COMSIG_PARENT_EXAMINE)
+
+/// Returns whether the game is supposed to consider the parent "secure".
+/datum/component/keep_me_secure/proc/is_secured()
+	var/obj/item/item_parent = parent
+	if (last_secured_location == get_turf(item_parent))
+		return FALSE
+
+	var/mob/holder = item_parent.pulledby || get(parent, /mob)
+	if (isnull(holder?.client))
+		return FALSE
+
+	return TRUE
+
+/datum/component/keep_me_secure/process(delta_time)
+	var/turf/current_turf = get_turf(parent)
+
+	if(is_secured())
+		last_secured_location = current_turf
+		last_move = world.time
+		if(secured_callback)
+			secured_callback.Invoke(last_move)
```

Comment:
Okay but for real, this probably shouldn't be operating on callbacks, it's against how we do components and signals, this should instead be sending a signal that the object's been secured so that things can listen in for securing(IE. an alarm that blares loudly every time the disk isn't secured)